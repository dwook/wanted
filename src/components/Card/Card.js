import React from 'react';
import './Card.scss';

function Card({
  company,
  address,
  title_img,
  position,
  reward,
  id,
  like_count
}) {
  return (
    <div className="card">
      <a href={`https://www.wanted.co.kr/wd/${id}`}>
        <div className="thumbnail">
          <img src={title_img.thumb} alt={company.name} />
          <div className="likeButton">{like_count}</div>
        </div>
        <div className="body">
          <div className="position">{position}</div>
          <div className="details">
            <div>{company.name}</div>
            <div className="address">
              <span>{address.location}</span>
              <span>{address.country}</span>
            </div>
            <div className="reward">채용보상금 {reward.formatted_total}</div>
          </div>
        </div>
      </a>
    </div>
  );
}

export default Card;
