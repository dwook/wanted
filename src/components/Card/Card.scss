.card {
  width: 25%;
  padding: 10px;
  list-style: none;
  display: inline-block;
  vertical-align: top;
  box-sizing: border-box;
  .thumbnail {
    position: relative;
    display: flex;
    border-radius: 4px;
    overflow: hidden;
    box-shadow: inset 0 0 0 1px rgba(0, 0, 0, 0.1);
    .likeButton {
      position: absolute;
      right: 10px;
      top: 10px;
      width: 60px;
      height: 30px;
      display: flex;
      align-items: center;
      justify-content: center;
      border-radius: 3px;
      background: rgba(0, 0, 0, 0.3);
      font-size: 12px;
      font-weight: 400;
      color: #fff;
    }
    img {
      width: 100%;
    }
  }
  .body {
    padding: 14px 10px;
  }
  .position {
    color: #333;
    font-size: 18px;
    font-weight: 600;
    line-height: 1.4;
    max-height: 2.8em;
    text-overflow: ellipsis;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
  }
  .details {
    margin-top: 8px;
    color: #999;
    font-size: 14px;
    font-weight: 400;
    line-height: 1.6;
  }
  .address {
    span:first-child::after {
      content: 'Â·';
      display: inline-block;
      padding: 0 2px;
    }
  }
  .reward {
    margin-top: 10px;
    color: #666;
    font-size: 13px;
    font-weight: 400;
  }
}
