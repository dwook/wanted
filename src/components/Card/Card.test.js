import React from 'react';
import { mount } from 'enzyme';
import Card from './Card';

describe('<Card />', () => {
  let component = null;
  const job = {
    status: 'active',
    is_like: false,
    is_bookmark: false,
    company: {
      id: 765,
      name: '마플(MARPPLE)',
      industry_name: 'IT, 컨텐츠'
    },
    title_img: {
      origin:
        'https://static.wanted.co.kr/images/company/765/6738_2_0.b211b4b3__1080_790.jpg',
      thumb:
        'https://static.wanted.co.kr/images/company/765/6738_2_0.b211b4b3__400_400.jpg'
    },
    compare_country: true,
    due_time: '2019-12-31',
    like_count: 42,
    logo_img: {
      origin:
        'https://static.wanted.co.kr/images/wdes/0_4_94e8e8c5-dfad-455e-bcd9-bf64b31d89cc.jpg',
      thumb:
        'https://static.wanted.co.kr/images/wdes/0_5_94e8e8c5-dfad-455e-bcd9-bf64b31d89cc.jpg'
    },
    address: {
      country: '한국',
      location: '경기'
    },
    position: '앱 개발(아이폰/안드로이드)',
    reward: {
      formatted_total: '1,000,000원',
      formatted_recommender: '500,000원',
      formatted_recommendee: '500,000원'
    },
    id: 18438
  };
  it('renders correctly', () => {
    component = mount(<Card {...job} />);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
