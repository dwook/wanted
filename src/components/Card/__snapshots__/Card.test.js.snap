// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> matches snapshot 1`] = `
<Card
  address={
    Object {
      "country": "한국",
      "location": "경기",
    }
  }
  company={
    Object {
      "id": 765,
      "industry_name": "IT, 컨텐츠",
      "name": "마플(MARPPLE)",
    }
  }
  compare_country={true}
  due_time="2019-12-31"
  id={18438}
  is_bookmark={false}
  is_like={false}
  like_count={42}
  logo_img={
    Object {
      "origin": "https://static.wanted.co.kr/images/wdes/0_4_94e8e8c5-dfad-455e-bcd9-bf64b31d89cc.jpg",
      "thumb": "https://static.wanted.co.kr/images/wdes/0_5_94e8e8c5-dfad-455e-bcd9-bf64b31d89cc.jpg",
    }
  }
  position="앱 개발(아이폰/안드로이드)"
  reward={
    Object {
      "formatted_recommendee": "500,000원",
      "formatted_recommender": "500,000원",
      "formatted_total": "1,000,000원",
    }
  }
  status="active"
  title_img={
    Object {
      "origin": "https://static.wanted.co.kr/images/company/765/6738_2_0.b211b4b3__1080_790.jpg",
      "thumb": "https://static.wanted.co.kr/images/company/765/6738_2_0.b211b4b3__400_400.jpg",
    }
  }
>
  <div
    className="card"
  >
    <a
      href="https://www.wanted.co.kr/wd/18438"
    >
      <div
        className="thumbnail"
      >
        <img
          alt="마플(MARPPLE)"
          src="https://static.wanted.co.kr/images/company/765/6738_2_0.b211b4b3__400_400.jpg"
        />
        <div
          className="likeButton"
        >
          42
        </div>
      </div>
      <div
        className="body"
      >
        <div
          className="position"
        >
          앱 개발(아이폰/안드로이드)
        </div>
        <div
          className="details"
        >
          <div>
            마플(MARPPLE)
          </div>
          <div
            className="address"
          >
            <span>
              경기
            </span>
            <span>
              한국
            </span>
          </div>
          <div
            className="reward"
          >
            채용보상금 
            1,000,000원
          </div>
        </div>
      </div>
    </a>
  </div>
</Card>
`;
